### ISSUE TITLE
Inaccurate records

### ISSUE THREAD

**Issue by Theepwner** [BanManagement/BanManager#216]
I initially reported this issue to the web panel repo, but was directed to report it here if the problem persisted on the latest builds.

I actually have been using the latest dev builds, and on build 91 since the day it was uploaded.

Now what's the bug?
Lets say I tempmuted a player named Bob with my account named Tom. It is shown perfectly in the web panel, and displayed accurately in the Previous Mutes section after the mute is over. But, after some time has passed that Previous Mute will swap the player that muted him with the player that unmuted him (console in this case).
So instead of being muted by Tom and unmuted by Console, it will say muted by Console and unmuted by Tom.
Real example: http://unchainedmc.net/bans/index.php?action=viewplayer&player=Wallstreet&server=0

Things get more complicated when you have multiple tempmute records.
http://unchainedmc.net/bans/index.php?action=viewplayer&player=UnchainedMC_Net&server=0
Take a look at the Previous Mutes section, you will see 5 Previous Mutes. The issue however, is that 2 of them are like duplicates, as well as experiencing the issue shown above.
Basically, I tempmuted the account UnchainedMC_Net 3 different times, shown by the mute reason "test", "test 2", and "test 3". I did not manually unmute the player to test, I simply let the mute run out before proceeding.

Hopefully these weird bugs can be ironed out soon. If you would like me to help, by using these current records and possibly testing more, I'd be glad to.

**Comment by confuser** [comment]
Have you got multiple servers syncing with the same local database?

**Comment by Theepwner** [comment]
Yes, and 2 different dedis.

**Comment by confuser** [comment]
Related:
```
2015-04-08 13:58:45 [WARN] java.sql.SQLException: Unable to run insert stmt on object me.confuser.banmanager.data.PlayerBanData@12c0b712: INSERT INTO `hub_bm_player_bans` (`player_id` ,`reason` ,`actor_id` ,`created` ,`updated` ,`expires` ) VALUES (?,?,?,?,?,?)
2015-04-08 13:58:45 [WARN] 	at me.confuser.banmanager.ormlite.misc.SqlExceptionUtil.create(SqlExceptionUtil.java:22)
2015-04-08 13:58:45 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:135)
2015-04-08 13:58:45 [WARN] 	at me.confuser.banmanager.ormlite.stmt.StatementExecutor.create(StatementExecutor.java:450)
2015-04-08 13:58:45 [WARN] 	at me.confuser.banmanager.ormlite.dao.BaseDaoImpl.create(BaseDaoImpl.java:310)
2015-04-08 13:58:45 [WARN] 	at me.confuser.banmanager.storage.PlayerBanStorage.ban(PlayerBanStorage.java:109)
2015-04-08 13:58:45 [WARN] 	at me.confuser.banmanager.runnables.ExternalBanSync.newBans(ExternalBanSync.java:70)
2015-04-08 13:58:45 [WARN] 	at me.confuser.banmanager.runnables.ExternalBanSync.run(ExternalBanSync.java:37)
2015-04-08 13:58:45 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftTask.run(CraftTask.java:71)
2015-04-08 13:58:45 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:53)
2015-04-08 13:58:45 [WARN] Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry '\x07I\xE3\x01\xFF\xB8JI\xBD\xC9M5\xB8"\x07!' for key 'hub_bm_player_bans_player_idx'
2015-04-08 13:58:45 [WARN] 	at com.mysql.jdbc.Util.handleNewInstance(Util.java:407)
2015-04-08 13:58:45 [WARN] 	at com.mysql.jdbc.Util.getInstance(Util.java:382)
2015-04-08 13:58:45 [WARN] 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1039)
2015-04-08 13:58:45 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3593)
2015-04-08 13:58:45 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3525)
2015-04-08 13:58:45 [WARN] 	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:1986)
2015-04-08 13:58:45 [WARN] 	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2140)
2015-04-08 13:58:45 [WARN] 	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2626)
2015-04-08 13:58:45 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:2111)
2015-04-08 13:58:45 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2407)
2015-04-08 13:58:45 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2325)
2015-04-08 13:58:45 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2310)
2015-04-08 13:58:45 [WARN] 	at me.confuser.banmanager.ormlite.jdbc.JdbcDatabaseConnection.insert(JdbcDatabaseConnection.java:170)
2015-04-08 13:58:45 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:91)
2015-04-08 14:00:55 [WARN] java.sql.SQLException: Unable to run insert stmt on object me.confuser.banmanager.data.PlayerBanData@6c853900: INSERT INTO `hub_bm_player_bans` (`player_id` ,`reason` ,`actor_id` ,`created` ,`updated` ,`expires` ) VALUES (?,?,?,?,?,?)
2015-04-08 14:00:55 [WARN] 	at me.confuser.banmanager.ormlite.misc.SqlExceptionUtil.create(SqlExceptionUtil.java:22)
2015-04-08 14:00:55 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:135)
2015-04-08 14:00:55 [WARN] 	at me.confuser.banmanager.ormlite.stmt.StatementExecutor.create(StatementExecutor.java:450)
2015-04-08 14:00:55 [WARN] 	at me.confuser.banmanager.ormlite.dao.BaseDaoImpl.create(BaseDaoImpl.java:310)
2015-04-08 14:00:55 [WARN] 	at me.confuser.banmanager.storage.PlayerBanStorage.ban(PlayerBanStorage.java:109)
2015-04-08 14:00:55 [WARN] 	at me.confuser.banmanager.commands.TempBanCommand$1.run(TempBanCommand.java:130)
2015-04-08 14:00:55 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftTask.run(CraftTask.java:71)
2015-04-08 14:00:55 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:53)
2015-04-08 14:00:55 [WARN] Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry '\x07I\xE3\x01\xFF\xB8JI\xBD\xC9M5\xB8"\x07!' for key 'hub_bm_player_bans_player_idx'
2015-04-08 14:00:55 [WARN] 	at com.mysql.jdbc.Util.handleNewInstance(Util.java:407)
2015-04-08 14:00:55 [WARN] 	at com.mysql.jdbc.Util.getInstance(Util.java:382)
2015-04-08 14:00:55 [WARN] 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1039)
2015-04-08 14:00:55 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3593)
2015-04-08 14:00:55 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3525)
2015-04-08 14:00:55 [WARN] 	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:1986)
2015-04-08 14:00:55 [WARN] 	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2140)
2015-04-08 14:00:55 [WARN] 	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2626)
2015-04-08 14:00:55 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:2111)
2015-04-08 14:00:55 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2407)
2015-04-08 14:00:55 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2325)
2015-04-08 14:00:55 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2310)
2015-04-08 14:00:55 [WARN] 	at me.confuser.banmanager.ormlite.jdbc.JdbcDatabaseConnection.insert(JdbcDatabaseConnection.java:170)
2015-04-08 14:00:55 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:91)
2015-04-08 14:04:45 [WARN] java.sql.SQLException: Unable to run insert stmt on object me.confuser.banmanager.data.PlayerBanData@58cdbe6a: INSERT INTO `hub_bm_player_bans` (`player_id` ,`reason` ,`actor_id` ,`created` ,`updated` ,`expires` ) VALUES (?,?,?,?,?,?)
2015-04-08 14:04:45 [WARN] 	at me.confuser.banmanager.ormlite.misc.SqlExceptionUtil.create(SqlExceptionUtil.java:22)
2015-04-08 14:04:45 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:135)
2015-04-08 14:04:45 [WARN] 	at me.confuser.banmanager.ormlite.stmt.StatementExecutor.create(StatementExecutor.java:450)
2015-04-08 14:04:45 [WARN] 	at me.confuser.banmanager.ormlite.dao.BaseDaoImpl.create(BaseDaoImpl.java:310)
2015-04-08 14:04:45 [WARN] 	at me.confuser.banmanager.storage.PlayerBanStorage.ban(PlayerBanStorage.java:109)
2015-04-08 14:04:45 [WARN] 	at me.confuser.banmanager.runnables.ExternalBanSync.newBans(ExternalBanSync.java:70)
2015-04-08 14:04:45 [WARN] 	at me.confuser.banmanager.runnables.ExternalBanSync.run(ExternalBanSync.java:37)
2015-04-08 14:04:45 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftTask.run(CraftTask.java:71)
2015-04-08 14:04:45 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:53)
2015-04-08 14:04:45 [WARN] Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry '\x07I\xE3\x01\xFF\xB8JI\xBD\xC9M5\xB8"\x07!' for key 'hub_bm_player_bans_player_idx'
2015-04-08 14:04:45 [WARN] 	at com.mysql.jdbc.Util.handleNewInstance(Util.java:407)
2015-04-08 14:04:45 [WARN] 	at com.mysql.jdbc.Util.getInstance(Util.java:382)
2015-04-08 14:04:45 [WARN] 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1039)
2015-04-08 14:04:45 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3593)
2015-04-08 14:04:45 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3525)
2015-04-08 14:04:45 [WARN] 	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:1986)
2015-04-08 14:04:45 [WARN] 	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2140)
2015-04-08 14:04:45 [WARN] 	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2626)
2015-04-08 14:04:45 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:2111)
2015-04-08 14:04:45 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2407)
2015-04-08 14:04:45 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2325)
2015-04-08 14:04:45 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2310)
2015-04-08 14:04:45 [WARN] 	at me.confuser.banmanager.ormlite.jdbc.JdbcDatabaseConnection.insert(JdbcDatabaseConnection.java:170)
2015-04-08 14:04:45 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:91)
2015-04-08 14:07:06 [WARN] java.sql.SQLException: Unable to run insert stmt on object me.confuser.banmanager.data.PlayerBanData@44fc8d7c: INSERT INTO `hub_bm_player_bans` (`player_id` ,`reason` ,`actor_id` ,`created` ,`updated` ,`expires` ) VALUES (?,?,?,?,?,?)
2015-04-08 14:07:06 [WARN] 	at me.confuser.banmanager.ormlite.misc.SqlExceptionUtil.create(SqlExceptionUtil.java:22)
2015-04-08 14:07:06 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:135)
2015-04-08 14:07:06 [WARN] 	at me.confuser.banmanager.ormlite.stmt.StatementExecutor.create(StatementExecutor.java:450)
2015-04-08 14:07:06 [WARN] 	at me.confuser.banmanager.ormlite.dao.BaseDaoImpl.create(BaseDaoImpl.java:310)
2015-04-08 14:07:06 [WARN] 	at me.confuser.banmanager.storage.PlayerBanStorage.ban(PlayerBanStorage.java:109)
2015-04-08 14:07:06 [WARN] 	at me.confuser.banmanager.commands.BanCommand$1.run(BanCommand.java:114)
2015-04-08 14:07:06 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftTask.run(CraftTask.java:71)
2015-04-08 14:07:06 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:53)
2015-04-08 14:07:06 [WARN] Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry '\x07I\xE3\x01\xFF\xB8JI\xBD\xC9M5\xB8"\x07!' for key 'hub_bm_player_bans_player_idx'
2015-04-08 14:07:06 [WARN] 	at com.mysql.jdbc.Util.handleNewInstance(Util.java:407)
2015-04-08 14:07:06 [WARN] 	at com.mysql.jdbc.Util.getInstance(Util.java:382)
2015-04-08 14:07:06 [WARN] 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1039)
2015-04-08 14:07:06 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3593)
2015-04-08 14:07:06 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3525)
2015-04-08 14:07:06 [WARN] 	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:1986)
2015-04-08 14:07:06 [WARN] 	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2140)
2015-04-08 14:07:06 [WARN] 	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2626)
2015-04-08 14:07:06 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:2111)
2015-04-08 14:07:06 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2407)
2015-04-08 14:07:06 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2325)
2015-04-08 14:07:06 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2310)
2015-04-08 14:07:06 [WARN] 	at me.confuser.banmanager.ormlite.jdbc.JdbcDatabaseConnection.insert(JdbcDatabaseConnection.java:170)
2015-04-08 14:07:06 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:91)
2015-04-08 14:07:14 [WARN] java.sql.SQLException: Unable to run insert stmt on object me.confuser.banmanager.data.PlayerBanData@4e40cb11: INSERT INTO `hub_bm_player_bans` (`player_id` ,`reason` ,`actor_id` ,`created` ,`updated` ,`expires` ) VALUES (?,?,?,?,?,?)
2015-04-08 14:07:14 [WARN] 	at me.confuser.banmanager.ormlite.misc.SqlExceptionUtil.create(SqlExceptionUtil.java:22)
2015-04-08 14:07:14 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:135)
2015-04-08 14:07:14 [WARN] 	at me.confuser.banmanager.ormlite.stmt.StatementExecutor.create(StatementExecutor.java:450)
2015-04-08 14:07:14 [WARN] 	at me.confuser.banmanager.ormlite.dao.BaseDaoImpl.create(BaseDaoImpl.java:310)
2015-04-08 14:07:14 [WARN] 	at me.confuser.banmanager.storage.PlayerBanStorage.ban(PlayerBanStorage.java:109)
2015-04-08 14:07:14 [WARN] 	at me.confuser.banmanager.commands.BanCommand$1.run(BanCommand.java:114)
2015-04-08 14:07:14 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftTask.run(CraftTask.java:71)
2015-04-08 14:07:14 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:53)
2015-04-08 14:07:14 [WARN] Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry '\x07I\xE3\x01\xFF\xB8JI\xBD\xC9M5\xB8"\x07!' for key 'hub_bm_player_bans_player_idx'
2015-04-08 14:07:14 [WARN] 	at com.mysql.jdbc.Util.handleNewInstance(Util.java:407)
2015-04-08 14:07:14 [WARN] 	at com.mysql.jdbc.Util.getInstance(Util.java:382)
2015-04-08 14:07:14 [WARN] 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1039)
2015-04-08 14:07:14 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3593)
2015-04-08 14:07:14 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3525)
2015-04-08 14:07:14 [WARN] 	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:1986)
2015-04-08 14:07:14 [WARN] 	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2140)
2015-04-08 14:07:14 [WARN] 	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2626)
2015-04-08 14:07:14 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:2111)
2015-04-08 14:07:14 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2407)
2015-04-08 14:07:14 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2325)
2015-04-08 14:07:14 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2310)
2015-04-08 14:07:14 [WARN] 	at me.confuser.banmanager.ormlite.jdbc.JdbcDatabaseConnection.insert(JdbcDatabaseConnection.java:170)
2015-04-08 14:07:14 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:91)
2015-04-08 14:07:21 [WARN] java.sql.SQLException: Unable to run insert stmt on object me.confuser.banmanager.data.PlayerBanData@3b7784f2: INSERT INTO `hub_bm_player_bans` (`player_id` ,`reason` ,`actor_id` ,`created` ,`updated` ,`expires` ) VALUES (?,?,?,?,?,?)
2015-04-08 14:07:21 [WARN] 	at me.confuser.banmanager.ormlite.misc.SqlExceptionUtil.create(SqlExceptionUtil.java:22)
2015-04-08 14:07:21 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:135)
2015-04-08 14:07:21 [WARN] 	at me.confuser.banmanager.ormlite.stmt.StatementExecutor.create(StatementExecutor.java:450)
2015-04-08 14:07:21 [WARN] 	at me.confuser.banmanager.ormlite.dao.BaseDaoImpl.create(BaseDaoImpl.java:310)
2015-04-08 14:07:21 [WARN] 	at me.confuser.banmanager.storage.PlayerBanStorage.ban(PlayerBanStorage.java:109)
2015-04-08 14:07:21 [WARN] 	at me.confuser.banmanager.commands.BanCommand$1.run(BanCommand.java:114)
2015-04-08 14:07:21 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftTask.run(CraftTask.java:71)
2015-04-08 14:07:21 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:53)
2015-04-08 14:07:21 [WARN] Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry '\x07I\xE3\x01\xFF\xB8JI\xBD\xC9M5\xB8"\x07!' for key 'hub_bm_player_bans_player_idx'
2015-04-08 14:07:21 [WARN] 	at com.mysql.jdbc.Util.handleNewInstance(Util.java:407)
2015-04-08 14:07:21 [WARN] 	at com.mysql.jdbc.Util.getInstance(Util.java:382)
2015-04-08 14:07:21 [WARN] 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1039)
2015-04-08 14:07:21 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3593)
2015-04-08 14:07:21 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3525)
2015-04-08 14:07:21 [WARN] 	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:1986)
2015-04-08 14:07:21 [WARN] 	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2140)
2015-04-08 14:07:21 [WARN] 	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2626)
2015-04-08 14:07:21 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:2111)
2015-04-08 14:07:21 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2407)
2015-04-08 14:07:21 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2325)
2015-04-08 14:07:21 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2310)
2015-04-08 14:07:21 [WARN] 	at me.confuser.banmanager.ormlite.jdbc.JdbcDatabaseConnection.insert(JdbcDatabaseConnection.java:170)
2015-04-08 14:07:21 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:91)
2015-04-08 14:16:34 [WARN] java.sql.SQLException: Unable to run insert stmt on object me.confuser.banmanager.data.PlayerBanData@4a3468dc: INSERT INTO `hub_bm_player_bans` (`player_id` ,`reason` ,`actor_id` ,`created` ,`updated` ,`expires` ) VALUES (?,?,?,?,?,?)
2015-04-08 14:16:34 [WARN] 	at me.confuser.banmanager.ormlite.misc.SqlExceptionUtil.create(SqlExceptionUtil.java:22)
2015-04-08 14:16:34 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:135)
2015-04-08 14:16:34 [WARN] 	at me.confuser.banmanager.ormlite.stmt.StatementExecutor.create(StatementExecutor.java:450)
2015-04-08 14:16:34 [WARN] 	at me.confuser.banmanager.ormlite.dao.BaseDaoImpl.create(BaseDaoImpl.java:310)
2015-04-08 14:16:34 [WARN] 	at me.confuser.banmanager.storage.PlayerBanStorage.ban(PlayerBanStorage.java:109)
2015-04-08 14:16:34 [WARN] 	at me.confuser.banmanager.commands.BanCommand$1.run(BanCommand.java:114)
2015-04-08 14:16:34 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftTask.run(CraftTask.java:71)
2015-04-08 14:16:34 [WARN] 	at org.bukkit.craftbukkit.v1_7_R4.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:53)
2015-04-08 14:16:34 [WARN] Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry '\x07I\xE3\x01\xFF\xB8JI\xBD\xC9M5\xB8"\x07!' for key 'hub_bm_player_bans_player_idx'
2015-04-08 14:16:34 [WARN] 	at com.mysql.jdbc.Util.handleNewInstance(Util.java:407)
2015-04-08 14:16:34 [WARN] 	at com.mysql.jdbc.Util.getInstance(Util.java:382)
2015-04-08 14:16:34 [WARN] 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1039)
2015-04-08 14:16:34 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3593)
2015-04-08 14:16:34 [WARN] 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3525)
2015-04-08 14:16:34 [WARN] 	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:1986)
2015-04-08 14:16:34 [WARN] 	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2140)
2015-04-08 14:16:34 [WARN] 	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2626)
2015-04-08 14:16:34 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:2111)
2015-04-08 14:16:34 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2407)
2015-04-08 14:16:34 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2325)
2015-04-08 14:16:34 [WARN] 	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2310)
2015-04-08 14:16:34 [WARN] 	at me.confuser.banmanager.ormlite.jdbc.JdbcDatabaseConnection.insert(JdbcDatabaseConnection.java:170)
2015-04-08 14:16:34 [WARN] 	at me.confuser.banmanager.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:91)
```

**Comment by Theepwner** [comment]
When (or if) you implement a fix, will it be able to repair/fix current data? It isn't a big deal, but it would be great if possible.

**Comment by confuser** [comment]
Probably not as it'd be difficult to differentiate between intentional duplicated data or not and may cause people to lose data they still want.

**Comment by confuser** [comment]
Need a way to reproduce this as at the moment it seems sporadic.

**Comment by Theepwner** [comment]
Sporadic on which build? On build 91 it happens consistently.

**Comment by confuser** [comment]
I've narrowed down the cause of this issue to the ExpiresSync runnable.

**Comment by confuser** [comment]
Please note this is an experimental fix. If you still experience this issue, please re-comment here and I will reopen.
